framework/include/base/MooseObjectWarehouseBase.h

```diff
@@ -539,6 +550,56 @@ MooseObjectWarehouseBase<T>::updateVariableDependencyHelper(std::set<MooseVariab
   }
 }
 
+template<typename T>
+void
+MooseObjectWarehouseBase<T>::updateMatPropDependency(std::set<unsigned int> & needed_mat_props, THREAD_ID tid/* = 0*/) const
+{
+  if (hasActiveObjects(tid))
+    updateMatPropDependencyHelper(needed_mat_props, _all_objects[tid]);
+}
+
+template<typename T>
+void
+MooseObjectWarehouseBase<T>::updateBlockMatPropDependency(SubdomainID id, std::set<unsigned int> & needed_mat_props, THREAD_ID tid/* = 0*/) const
+{
+  if (hasActiveBlockObjects(id, tid))
+    updateMatPropDependencyHelper(needed_mat_props, getActiveBlockObjects(id, tid));
+}
+
+
+template<typename T>
+void
+MooseObjectWarehouseBase<T>::updateBoundaryMatPropDependency(std::set<unsigned int> & needed_mat_props, THREAD_ID tid/* = 0*/) const
+{
+  if (hasActiveBoundaryObjects(tid))
+  {
+    typename std::map<BoundaryID, std::vector<MooseSharedPointer<T> > >::const_iterator it;
```

Comment:
This is OK and I know it's copying from other code but we can just use `auto` now for these long declarations. In fact, you can just use a range based for loop here to make this really clean and tight!